R/kelp.R

```diff
@@ -0,0 +1,110 @@
+#' Create a kelp object for interacting with SeaweedFS.
+#'
+#' This is a slightly higher level abstraction than [seaweed_master()] and
+#' [seaweed_volume()].
+#'
+#' @keywords export
+#' @importFrom R6 R6Class
+kelp <- R6::R6Class(
+  "kelp",
+  cloneable = FALSE,
+
+  public = list(
+    #' @field master A [seaweed_master] object for sending requests
+    master = NULL,
+
+    #' @description
+    #' Create client object for sending http requests to seaweed
+    #'
+    #' @param seaweed_url Root URL of Seaweed
+    #'
+    #' @return A new `kelp` object
+    initialize = function(seaweed_url) {
+      self$master <- seaweed_master$new(seaweed_url)
+    },
+
+    #' @description
+    #' Upload file to SeaweedFS
+    #'
+    #' @param path Path to file to be uploaded
+    #' @param collection Collection name, acts as a namespace for files.
+    #'
+    #' @return The uploaded file ID.
+    upload_file = function(path, collection = NULL) {
+      key <- self$master$assign(collection = collection)
+      volume <- seaweed_volume$new(key$publicUrl)
+      volume$upload_file(key$fid, path)
+      key$fid
+    },
+
+    #' @description
+    #' Download file from SeaweedFS
+    #'
+    #' @param fid SeaweedFS file ID to read
```

Comment:
I wonder about using `id` rather than `fid`, especially if we're not always storing files

Reply:
Yeah that sounds good, the `kelp` class doesn't have any functions which work with volume ids either so not much scope for confusion between types. Have updated this